jwst/datamodels/tests/test_models.py

```diff
@@ -615,12 +615,18 @@ def test_info():
 
 
 def test_validate_on_read():
+    """ Test for proper validation error
+
+    Note: The FITS file is opened separately in order to properly close
+    the file.
+    """
     schema = ImageModel((10, 10))._schema.copy()
     schema['properties']['meta']['properties']['calibration_software_version']['fits_required'] = True
+    hduls = fits.open(FITS_FILE)
```

Comment:
In this case I think using the context manager should work right?

```python
with fits.open(FITS_FILE) as hduls:
    with pytest.raises(jsonschema.ValidationError):
        ImageModel(hduls, schema=schema, strict_validation=True)
```

Reply:
If I remember correctly, I did try that and it didn't work, in the sense that the pytest openfiles plugin still failed. Will try again just to make sure.

Reply:
Well, it did work this time. Something else must have been in play at the time. 👍

Reply:
Weird!